grpc-api/src/main/java/io/deephaven/grpc_api/session/SessionState.java

```diff
@@ -733,23 +744,38 @@ private synchronized ExportNotification makeExportNotification() {
             if (errorId != null) {
                 builder.setContext(errorId);
             }
-            if (dependentErrorId != null) {
-                builder.setDependentHandle(dependentErrorId);
+            if (dependentHandle != null) {
+                builder.setDependentHandle(dependentHandle);
             }
 
             return builder.build();
         }
     }
 
     public void addExportListener(final StreamObserver<ExportNotification> observer) {
+        if (isExpired()) {
+            throw GrpcUtil.statusRuntimeException(Code.UNAUTHENTICATED, "session has expired");
+        }
+
         final ExportListener listener = new ExportListener(observer);
         manage(listener);
```

Comment:
I think it makes more sense for `SessionState` to not be a `LivenessArtifact` and use `retainReference` and `dropReference` rather than `manage`/`unmanage`.